src/commands/pickFuncProcess.ts

```diff
@@ -73,69 +73,87 @@ async function startFuncTask(context: IActionContext, workspaceFolder: vscode.Wo
     try {
         await vscode.tasks.executeTask(funcTask);
 
+        const intervalMs: number = 500;
+        const client: ServiceClient = await createGenericClient();
+        const statusRequest: WebResource = getStatusRequest(funcTask, intervalMs);
         const maxTime: number = Date.now() + timeoutInSeconds * 1000;
         while (Date.now() < maxTime) {
             if (taskError !== undefined) {
                 throw taskError;
             }
 
             const taskInfo: IRunningFuncTask | undefined = runningFuncTaskMap.get(workspaceFolder);
-            // Ensure func task runs for at least 1 second in case it fails on startup and thus `taskError` will be thrown instead
-            if (taskInfo && Date.now() > taskInfo.startTime + 1000) {
-                return taskInfo.processId.toString();
+            if (taskInfo) {
+                try {
+                    // wait for status url to indicate functions host is running
+                    await client.sendRequest(statusRequest);
```

Comment:
https://github.com/Azure/azure-functions-core-tools/issues/185#issuecomment-315484240

Looks like you could check the state === 'Running' as well.  Not really sure, but I imagine you could have instances where it returns 200, but is still initializing?